modin/core/dataframe/pandas/exchange/dataframe_protocol/from_dataframe.py

```diff
@@ -0,0 +1,463 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Module houses functions building a ``pandas.DataFrame`` from DataFrame exchange protocol object."""
+
+import pandas
+import ctypes
+import numpy as np
+
+from typing import Optional, Tuple, Any, Union
+from modin.core.dataframe.base.exchange.dataframe_protocol.utils import (
+    DTypeKind,
+    ColumnNullType,
+)
+from modin.core.dataframe.base.exchange.dataframe_protocol.dataframe import (
+    ProtocolDataframe,
+    ProtocolColumn,
+    ProtocolBuffer,
+)
+
+
+def from_dataframe_to_pandas(
+    df: ProtocolDataframe, allow_copy: bool = True, nchunks: Optional[int] = None
+):
+    """
+    Build ``pandas.DataFrame`` from an object supporting DataFrame exchange protocol (__dataframe__).
+
+    Parameters
+    ----------
+    df : ProtocolDataframe
+        Object supporting the exchange protocol (__dataframe__).
+    allow_copy : bool, default: True
+        Whether to allow for `df` providing a copy of underlying data.
+    nchunks : int, optional
+        Number of chunks to split `df`.
+
+    Returns
+    -------
+    pandas.DataFrame
+    """
+    if not hasattr(df, "__dataframe__"):
+        raise ValueError("`df` does not support __dataframe__")
+
+    df = df.__dataframe__()
+    if isinstance(df, dict):
```

Comment:
I think we can remove this.

Reply:
Going by the current protocol specification, [`__dataframe__()` returns a dictionary](https://data-apis.org/dataframe-protocol/latest/API.html#:~:text=return%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%22dataframe%22%3A%20self%2C%20%20%23%20DataFrame%20object%20adhering%20to%20the%20protocol%0A%20%20%20%20%20%20%20%20%20%20%20%20%22version%22%3A%200%20%20%20%20%20%20%20%20%23%20Version%20number%20of%20the%20protocol%0A%20%20%20%20%20%20%20%20%7D). Although we don't do so in our implementations, some other libraries may strictly follow the spec, so I believe it's better to keep this simple check in order to cover more implementations.

Reply:
I think returning a dict will be replaced to returning a dataframe exactly in the future. Let's leave as is for now. We can remove this any time.